Mathlib/Analysis/InnerProductSpace/Positive.lean

```diff
@@ -206,6 +206,98 @@ end PartialOrder
 
 end ContinuousLinearMap
 
+/-- a star projection is an idempotent and self-adjoint element -/
+structure IsStarProjection {M : Type*} [Mul M] [Star M] (p : M) : Prop where
+  isIdempotentElem : IsIdempotentElem p
+  isSelfAdjoint : IsSelfAdjoint p
```

Comment:
I think personally I would rather see:
```suggestion
structure IsStarProjection {M : Type*} [Mul M] [Star M] (p : M) : Prop where
  protected mul_self : p * p = p
  protected star_eq : star p = p
```
and then
```lean
lemma IsStarProjection.isSelfAdjoint (hp : IsStarProjection p) : IsSelfAdjoint p := hp.star_eq

lemma IsStarProjection.isIdempotentElem (hp : IsStarProjection p) : IsIdempotentElem p := hp.mul_self
```
because then when you go to *prove* `IsStarProjection p`, you don't have to do any rewriting, but you still have access to the versions with dot notation when you have it as a hypothesis.

And yes, this should definitely be somewhere down low like `Algebra.Star.SelfAdjoint` or something (I don't remember the file structure).